network/src/logging.rs

```diff
@@ -1,34 +1,62 @@
 // Copyright (c) The Libra Core Contributors
 // SPDX-License-Identifier: Apache-2.0
 
-use crate::{
-    peer_manager::{ConnectionNotification, ConnectionRequest, PeerManagerRequest},
-    ConnectivityRequest,
-};
+//!
+//! This module is to contain all networking logging information.
+//!
+//! ```
+//! use libra_logger::prelude::*;
+//! use crate::logging::*;
+//!
+//! send_struct_log!(
+//!   network_log(network_events::CONNECTIVITY_MANAGER_LOOP)
+//!     .data(network_events::TYPE, network_events::START)
+//!     .field(network_events::EVENT_ID, 5)
+//! );
+//! ```
+//!
+
 use libra_config::network_id::NetworkContext;
-use libra_logger::LoggingField;
-use libra_types::PeerId;
+use libra_logger::StructuredLogEntry;
+
+/// A helper function to cut down on a bunch of repeated network struct log code
+pub fn network_log(label: &'static str, network_context: &NetworkContext) -> StructuredLogEntry {
+    StructuredLogEntry::new_named(label)
+        .data("category", "network")
```

Comment:
This is interesting. I originally thought we won't need category at all and can just use auto-identified crate name.
But at the same time, I guess single 'logical' category can span across multiple crates, so perhaps category is needed.

In this case, I would modify StructuredLogEntry and add `category:Option<&'static str>` and then change new_named to `new_named(category: &'static str, name: &'static str)`, e.g. basically always require category on named log entries.

Do you mind including this small refactoring in your pr(I can also do it myself, if you prefer me to do it)

Reply:
so we should probably move security to a category as well : o

Reply:
Sure I'll take a look into it.

My thoughts on how this would work changed over time.  I was confused why there was a "name" field, until I realized that we want to probably label groupings of logs.  Then I realized, that as a whole we should probably categorize them.

We could use the crate name, but I just figured this would make it simple in something like Kibana to filter "category"="network" or "name"="connectivity_manager_loop".

Thoughts?

Reply:
Yeah, sounds like it I'll go throw in security as well with a category.

Reply:
Yeah, I think we should just make a category field required.
After this we should also remove 'security' as a severity level.
Instead, for security logs we should have category=security and severity=critical